scipy/stats/_multicomp.py

```diff
@@ -0,0 +1,253 @@
+from __future__ import annotations
+
+from dataclasses import dataclass
+from typing import TYPE_CHECKING
+
+import numpy as np
+
+from scipy import stats
+from scipy.stats._qmc import check_random_state
+
+if TYPE_CHECKING:
+    import numpy.typing as npt
+    from scipy._lib._util import SeedType
+    from typing import Literal
+
+
+__all__ = [
+    'dunnett'
+]
+
+
+@dataclass
+class DunnettResult:
+    statistic: np.ndarray
+    pvalue: np.ndarray
+
+
+def dunnett(
+    *samples: npt.ArrayLike, control: npt.ArrayLike,
+    alternative: Literal['two-sided', 'less', 'greater'] = "two-sided",
+    random_state: SeedType = None
+) -> DunnettResult:
+    """Dunnett's test.
+
+    Parameters
+    ----------
+    sample1, sample2, ... : 1D array_like
+        The sample measurements for each experiment group.
```

Comment:
I have a significant preference for passing a list of sample arrays rather than using `*` argument expansion. I think most cases where I personally will use this, I'll always have a list of indeterminate length rather than multiple explicitly named variables. But even in the cases that I do, I'd like to read the function call with an explicit list. Mentally, my code-reading brain knows what's going where more immediately.

```python
samples = [x, y, z]

# Prefer
dunnett(samples, control=control)
dunnett([x, y, z], control=control)

# instead of
dunnett(*samples, control=control)
dunnett(x, y, z, control=control)
```

Reply:
I tend to agree, but I'm not convinced we should break the precedent set by `f_oneway`, `alexandergovern`, `kruskal`, `tukey_hsd`, `friedmanchisquare`....

Reply:
Fair.